src/BuildScriptGenerator/DotNetCore/DotnetCorePlatform.cs

```diff
@@ -112,6 +114,14 @@ public PlatformDetectorResult Detect(RepositoryContext context)
             BuildScriptGeneratorContext context,
             PlatformDetectorResult detectorResult)
         {
+            var dotNetCorePlatformDetectorResult = detectorResult as DotNetCorePlatformDetectorResult;
+            if (dotNetCorePlatformDetectorResult == null)
+            {
+                throw new ArgumentException(
+                    $"Expected '{nameof(detectorResult)}' argument to be of type " +
+                    $"'{typeof(DotNetCorePlatformDetectorResult)}' but got '{detectorResult.GetType()}'.");
```

Comment:
Do we want to directly throw `ArgumentException ` or use more user friendly exceptions like `InvalidException`

Reply:
`InvalidUsageException` is only to expose details to end users. The exception scenario here is something that would not be triggered by an end user since this area of code is used by our own types (ex: DefaultBuildScriptGenerator).